### PR TITLE
CDAP-8624 Load txPruneDebug class only when those endpoints are invoked

### PR DESCRIPTION
**Pull request by gokulavasan**
Also, disable pruning CDAP Standalone.

JIRA : https://issues.cask.co/browse/CDAP-8624

Build : http://builds.cask.co/browse/CDAP-RUT654

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 1]
```diff
@@ -292,4 +279,27 @@ public void getIdleRegions(HttpRequest request, HttpResponder responder,
       LOG.debug("Exception while trying to fetch the idle regions.", e);
     }
   }
+
+  private void initializePruningDebug() {
```
Comment (poornachandra) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: This method will need synchronization

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 2]
```diff
@@ -292,4 +279,27 @@ public void getIdleRegions(HttpRequest request, HttpResponder responder,
...
+    if (debugClazz == null || debugObject == null) {
+      try {
+        this.debugClazz = getClass().getClassLoader()
+          .loadClass("org.apache.tephra.hbase.txprune.InvalidListPruningDebug");
+        this.debugObject = debugClazz.newInstance();
+        Configuration hConf = new Configuration();
+        for (Map.Entry<String, String> entry : cConf) {
+          hConf.set(entry.getKey(), entry.getValue());
+        }
+        Method initMethod = debugClazz.getMethod("initialize", Configuration.class);
```
Comment (poornachandra) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: When looking back at Tephra's InvalidListPruningDebug class, I notice that the connection created in intialize is not closed. We need to add a destroy to InvalidListPruningDebug class, and call that method in destroy of this class.

**Review by poornachandra — COMMENTED** [review]
@gokulavasan A couple of comments

**Comment by poornachandra** [comment]
Also would be good to send a bad request when these methods are called and pruning is not enabled.

**Comment by gokulavasan** [comment]
@poornachandra Please review the last two commits. Thanks!

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java [hunk 3]
```diff
@@ -280,25 +279,50 @@ public void getIdleRegions(HttpRequest request, HttpResponder responder,
...
-        this.debugClazz = null;
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public void destroy(HandlerContext context) {
+    super.destroy(context);
+    if (debugClazz != null && debugObject != null) {
```
Comment (poornachandra) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/TransactionHttpHandler.java]: This also needs synchronization.

**Comment by poornachandra** [comment]
@gokulavasan Just one comment Gokul

**Comment by poornachandra** [comment]
LGTM

**Review by poornachandra — APPROVED** [review]
Please rebase and merge

**Comment by gokulavasan** [comment]
Rebased. Build is green. Merging.

**Pull request merged**